src/silverlabnwb/signature.py

```diff
@@ -220,12 +272,16 @@ def attrs_sig(self, entity):
         :param entity: an h5py group or dataset
         :returns: a string with the signature, one line per attribute
         """
-        sig = u'\n'.join(
-            [u'\t@{}: {}'.format(name, self.attr_val(value))
-             for name, value in sorted(entity.attrs.items())
-             if not self.ignored_attr(entity.name, name)
-             ])
-        return u'\n' + sig if sig else u''
+        sig = u''
+        for name, value in sorted(entity.attrs.items()):
+            if not self.ignored_attr(entity.name, name):
+                corrected_type = None
+                if hasattr(value, 'dtype'):
+                    corrected_type = self.should_cast_path(entity.name+'/'+name, value.dtype)
```

Comment:
I personally find this spacing clearer but opinions may vary...
```suggestion
                    corrected_type = self.should_cast_path(entity.name + '/' + name, value.dtype)
```

Reply:
That would be better according to pep8!

Reply:
But not the flake8 tool, apparently!